### PR TITLE
Use SDL_Renderer in SDL2 port

### PR DESCRIPTION
**Pull request by juliusikkala**
So, I noticed that the SDL2 port is extremely slow when there's fullscreen updates going on (e.g. change a color that's being used by every character on screen). A large window, through either a big font or just lots of rows & columns, would slow programs down to a crawl on my PC in such circumstances. So, in this draft PR, I'm proposing to do two things to fix this:

1. Implement a glyph cache
    - Previously, TTF_RenderGlyph* functions were getting called really often, practically again for every drawn character, which is extremely slow
    - Now, rendered glyphs are cached, so re-draws of the same character go much faster (even if the colors change! SDL_SetTextureColorMod() is used to achieve this)
    - The table I used for the glyph cache is naive. A hash table would consume less memory, but is harder to implement in C ;)
2. Switch to using SDL_Renderer & SDL_Textures
    - The previous SDL_Surface + SDL_BlitSurface combo forced software rendering, which quickly gets quite slow
    - SDL_Renderer allows using the GPU for rendering, and SDL_Textures replace SDL_Surfaces for that use case.

What's not cool about this PR (and which is why I submit this as a draft):

- Removes support for `PDC_SDL_RENDER_SHADED` (but `SOLID` & `BLENDED` are kept)
- Removes support for tiled background images

These are all features that _could_ be implemented with some elbow grease, but I'm feeling too lazy now to implement features that I'm currently not using myself (tiles, bitmap fonts). One particularly big issue is the deletion of the PDC_WIDE stuff from everywhere in the port. I did that because it made it easier to focus on the parts that I use myself, but isn't great for this PR itself.

### CODE REVIEW

File: sdl2/sdltest.c [hunk 1]
```diff
@@ -1,91 +0,0 @@
-/* Here's a simple example of combining SDL and PDCurses functionality.
```
Comment (GitMensch) [inline sdl2/sdltest.c:1]: I'd assume that more tests are added, not a sdl2-only test be removed?!?

File: sdl2/pdcsetsc.c [hunk 2]
```diff
@@ -92,14 +92,10 @@ int PDC_set_bold(bool boldon)
     if (!SP)
         return ERR;
 
-#ifdef PDC_WIDE
```
Comment (GitMensch) [inline sdl2/pdcsetsc.c:95]: Can you please try to rework the PR letting as much !PDC_WIDE code in as possible (this for example has zero effect, so definitely should be kept in)?
Doing so should keep the diff actually smaller and makes reviewing the patch easier.
I guess that in the end there won't be much questionable !PDC_WIDE parts - in this case we can work on those and drop one of the big "con" issues of this PR.

**Review by GitMensch — COMMENTED** [review]
What is the reason for switching to 16-bit chtypes?

**Comment by juliusikkala** [comment]
> What is the reason for switching to 16-bit chtypes?

I just botched the README, that's all. My own branches are a bit of a mess. I haven't actually done anything like that unless I botched even more things.

**Comment by juliusikkala** [comment]
I managed to get it working without PDC_WIDE, restoring support for the bitmap fonts. AFAIK, palettes unfortunately don't work with SDL_Textures, so colors would have to be supported in a different way. Possibly the same approach as with TTF fonts.

**Comment by juliusikkala** [comment]
In the end, the same bottleneck occurs in this approach as well. Just changing colors leads into thousands of draw calls as the SDL_Renderer API is unable to batch the draws. This chokes the GPU driver and ruins performance. It's still somewhat faster than the software rendering approach, but not Good Enough. So, I started making an OpenGL port instead: https://github.com/juliusikkala/PDCursesMod/tree/gl-port

I'll open a separate PR sometime later if I get it to a generally usable state. It already deals with my kinda obscure use-case, but doesn't work with attributes and the cursor seems bugged out. But at least it's hella fast because it only does one drawcall per frame!

**Comment by GitMensch** [comment]
Thanks! I'm looking forward to see the PR for OpenGL!

**Comment by Bill-Gray** [comment]
If we _do_ end up with both  SDL2 and SDL2/OpenGL platforms,  speeding up the former may still be of interest.  I tried the glyph caching,  and got about a 50% speedup in the `speed` test.  Not bad,  for not many lines of code.  It does seem as if the second part you tried out (using `SDL_Renderer` and `SDL_Textures`) may be more trouble than it's worth.

**Pull request closed without merging**